lib/beacon-verification.ts

```diff
@@ -48,13 +53,19 @@ function normP2(point: G2Hex): PointG2 {
     return point instanceof PointG2 ? point : PointG2.fromHex(point);
 }
 
-async function normP1Hash(point: G1Hex): Promise<PointG1> {
-    return point instanceof PointG1 ? point : PointG1.hashToCurve(point);
+async function normP1Hash(point: G1Hex, domainSeparationTag: string): Promise<PointG1> {
+    return point instanceof PointG1 ? point : PointG1.hashToCurve(point, {DST: domainSeparationTag});
 }
 
-export async function verifySigOnG1(signature: G1Hex, message: G1Hex, publicKey: G2Hex): Promise<boolean> {
+export async function verifySigOnG1(
+    signature: G1Hex,
+    message: G1Hex,
+    publicKey: G2Hex,
+    // default DST is the invalid one used for 'bls-unchained-on-g1' for backwards compat
+    domainSeparationTag= 'BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_NUL_'
```

Comment:
Gosh, I hate this. But I guess it's the only way not to break things for people T.T
Are we sure people are actually using this function directly?
Could we avoid doing this? In the code above we could be explicit for each scheme...

Reply:
we could make it mandatory, but we'd have to bump to v2 to make sure nobody gets pwned by it

Reply:
Have raised an issue here: https://github.com/drand/drand-client/issues/70